tests/test_hierarchy.cc

```diff
@@ -597,3 +597,65 @@ BOOST_AUTO_TEST_CASE(fast_multiply_transpose_mf)
         BOOST_TEST(fast_matrix->el(i, j) == ref_matrix->el(i, j),
                    tt::tolerance(1e-9));
 }
+
+template <int dim>
+class ConstantMaterialProperty;
+template <int dim>
+class LinearMaterialProperty;
+template <int dim>
+class LinearXMaterialProperty;
+template <int dim>
+class DiscontinuousMaterialProperty;
```

Comment:
Why did you add these declarations?  Aren't they already available in the header `test_hierarchy_helpers.hpp`?

Reply:
Ah, yes they made debugging easier. I'll remove them.